src/test/java/org/mule/service/http/impl/NtlmMockResponseGenerator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.http.impl;
+
+import static java.lang.ClassLoader.getSystemClassLoader;
+import static org.apache.http.client.config.AuthSchemes.NTLM;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.OK;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.UNAUTHORIZED;
+import static org.mule.runtime.http.api.HttpHeaders.Names.AUTHORIZATION;
+import static org.mule.runtime.http.api.HttpHeaders.Names.WWW_AUTHENTICATE;
+import static org.slf4j.LoggerFactory.getLogger;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+import org.mule.runtime.http.api.domain.message.response.HttpResponseBuilder;
+
+import java.io.FileInputStream;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+
+/**
+ * Test NTLM authentication response generator
+ */
+public class NtlmMockResponseGenerator {
+
+  private final Logger LOGGER = getLogger(NtlmMockResponseGenerator.class);
+
+  public enum State {
+    NOT_INITIATED, AUTH_REQUESTED, CHALLENGE_SENT, SUCCESS, FAILURE
+  }
+
+  private final String firstMessageExpectedHeader;
+  private final String challenge;
+  private final String secondMessageExpectedHeader;
+
+  private final String username;
+  private final String password;
+  private final String domain;
+
+  private State currentState;
+
+  private final static String DEFAULT_CREDENTIALS_FILE = "ntlm-credentials.conf";
```

Comment:
but why? do you change these anywhere?

Reply:
They are reused between two test suites, and in it seemed like a good idea to reuse them if some others NTLM tests are added.